src/app/clusters/scenes/SceneTableImpl.cpp

```diff
@@ -491,14 +482,20 @@ struct FabricSceneData : public PersistentData<kPersistentFabricBufferMax>
         TLV::TLVReader reader;
         reader.Init(buffer, size);
 
-        ReturnErrorOnFailure(DeserializeAdjust(reader, storage, deleted_scenes));
-        // If DeserializeAdjust sets the "deleted_scenes" variable, the table in flash memory held to many scenes (Can happend if
-        // max_scenes_per_fabric was reduced during an OTA) and was adjusted during deserailizing . The fabric data must then be
-        // updated
-        if (deleted_scenes)
+        err = Deserialize(reader, storage, deleted_scenes_count);
+
+        // If DeserializeAdjust sets the "deleted_scenes" variable, the table in flash memory held to many scenes (Can happend
+        // if max_scenes_per_fabric was reduced during an OTA) and was adjusted during deserailizing . The fabric data must then
+        // be updated
+        if (deleted_scenes_count)
         {
+            GlobalSceneCount global_count;
+            ReturnErrorOnFailure(global_count.Load(storage));
+            global_count.count_value = static_cast<uint8_t>(global_count.count_value - deleted_scenes_count);
+            ReturnErrorOnFailure(global_count.Save(storage));
             return this->Save(storage);
```

Comment:
So if Deserialize failed but this Save succeeded, we will return success?  That seems strange.

Reply:
Now returns err instead.